Zend/zend_long.h

```diff
@@ -23,7 +23,7 @@
 #include <stdint.h>
 
 /* This is the heart of the whole int64 enablement in zval. */
-#if defined(__x86_64__) || defined(__LP64__) || defined(_LP64) || defined(_WIN64)
+#if !defined(__APPLE__) && (defined(__x86_64__) || defined(__LP64__) || defined(_LP64) || defined(_WIN64))
```

Comment:
Ugh, it would be pretty sad, if we couldn't have 64bit integers on macOS.

Reply:
Yes I know, not too "proud" about it :-) ... but either we "ignore" it if we re sure it works well in fact in real hardware, mac Intel is a dying trend but will take a certain time to actually die.

Reply:
I'd assume the vast majority of people use a 64-bit build of PHP and might thus implicitly be depending on 64-bit integers. @devnexen Just disabling `zend_string_equal_val` for macOS doesn't do it?

Reply:
how for sure it builds if I do what I say but ... the main "issue" still remain then 

<img width="1245" alt="image" src="https://user-images.githubusercontent.com/4922778/181596806-6d98cb1b-3a88-4e4e-a747-4d684649ef63.png">

Reply:
Hmm, what is the diff? In #9087 it is:
````diff
     sscanf 64-bit signed int '9223372036854775807'  (2^63)-1 = 9223372036854775807
     sscanf 64-bit unsign int '18446744073709551615' (2^64)-1 = 18446744073709551615
     printf 64-bit signed int '9223372036854775807'  (2^63)-1 = 9223372036854775807
007+ printf 64-bit signed int '18446744073709551615' (2^64)-1 = 9223372036854775808
007- printf 64-bit signed int '18446744073709551615' (2^64)-1 = 0
     Done
````

That doesn't look like a *general* 64bit integer issue; and frankly, that part of the test looks fishy anyway.

Reply:
anyhow

> but either we "ignore" the failed test

@zeriyoshi said the test pass w/o issue in his intel mac.

Reply:
Do we support ARM for mac? Does it work without Rosetta?

Reply:
The point is that I would like to understand *why* this test fails on some platforms, and also, *why* it passes on (most) others. It doesn't appear to be related to the size of integers, since `18446744073709551615` is supposed to be parsed as float anyway.

Reply:
> Do we support ARM for mac? Does it work without Rosetta? It's not listed in https://wiki.php.net/platforms at least.

It does pass w/o issue on pure arm mode then my fix is not appropriate indeed.